### ISSUE TITLE
[BUG]: Attempting to revert uninstallation Windows services fails

### ISSUE THREAD

**Issue by genderneutralnoun** [undergroundwires/privacy.sexy#270]
### Description

Attempting to revert uninstallation of Your Phone Companion fails.

### OS

Windows 10 22h2

### Reproduction steps

1. Open privacy.sexy and click Revert for Your Phone Companion, and run the script.
2. Script errors out.

### Scripts

Was using the desktop application, but this should be identical.
[privacy-script.txt](https://github.com/undergroundwires/privacy.sexy/files/12848891/privacy-script.txt)

### Screenshots

No screenshots taken.

### Additional information

Also happens with Communications - Phone app, but I don't know if that's required for using Your Phone (as currently, even though reverting Your Phone app i successful, the app doesn't work). If I can find a good desktop app that lets me send texts easily, I'd be happy to drop Your Phone for it, but so far my search for one has been unsuccessful.

**Comment by genderneutralnoun** [comment]
Oops, forgot to add a title. Sorry.

**Comment by undergroundwires** [comment]
Hi @genderneutralnoun ,

Thanks for the report. I improved the reversion of Windows app as part of #260. It will be released at the end of this week, I will ping you when it's up and you can give it a try and let me know if it solves the issue for you, in that case we can close this one.

Just for your knowledge, both apps are missing on latest versions of Windows (22H2). But privacy.sexy new version will still be able to install those for you.

**Comment by undergroundwires** [comment]
So privacy.sexy [0.12.5](https://github.com/undergroundwires/privacy.sexy/releases/tag/0.12.5) with improved app revert logic is released. I tested it and see that it cannot install:

- Your Phone Companion (`Microsoft.WindowsPhone` and `Microsoft.Windows.Phone`)
- Communications - Phone (`Microsoft.CommsPhone`)

This is because Microsoft does no longer ship these packages and simply dropped them (see [blog post](https://support.microsoft.com/en-us/topic/introducing-microsoft-phone-link-and-link-to-windows-2e4bb4c0-f99a-4464-92a8-5264c7c39734 )). privacy.sexy is not a package manager, but has ability restore to default OS state after doing a change, but the default OS state does not include them anymore. If you'd use a legacy Windows version that included those apps, privacy.sexy would be able to revert those.

However it can revert `Microsoft.YourPhone` (Phone Link app). Phone Link replaces formally known "Your Phone", but it's essentially a new brand of the same package. See [store page](https://apps.microsoft.com/detail/phone-link/9NMPJ99VJBWV?hl=en-us&gl=us).

**Comment by genderneutralnoun** [comment]
Thank you for the explanation! I think I should be able to fix this on my own now. Should I let you know if I get it working again and explain the steps I took for anyone who has this problem in the future?

**Comment by undergroundwires** [comment]
Hi @genderneutralnoun , sorry for the late response. This would help others wondering the same thing in google searches. So do that if you'd want to help others :)

I'm adding this message to reverting apps to make this more clear:

```ps1
throw @"
Unable to reinstall the requested package (`"$packageName`").
It appears to no longer be included in this version of Windows.
You may search for it or an alternative in the Microsoft Store or consider using an earlier version of Windows where this
package was originally provided.
 "@
``

**Comment by undergroundwires** [comment]
User-friendly message and improved revert code is released in [0.12.6](https://github.com/undergroundwires/privacy.sexy/releases/tag/0.12.6) 🚀
